### DISCUSSION TITLE
How to upgrade a wix burn Custom bootstrapping application to Wix 4 rc2?

### DISCUSSION THREAD

**Discussion by woscostametaxa** [wixtoolset/issues#7216] [Questions] [answered]
Currently in my ManagedBootstrappingApplication.csproj with WiX 3.11.2 i reference it like this:
```xml    
<Reference Include="BootstrapperCore, Version=3.0.0.0, Culture=neutral, PublicKeyToken=ce35f76fcda82bad, processorArchitecture=MSIL">
  <SpecificVersion>False</SpecificVersion>
  <HintPath>..\..\..\..\Program Files (x86)\WiX Toolset v3.11\SDK\BootstrapperCore.dll</HintPath>
</Reference>
``` 
And in my CustomBootstrapper.cs (the one implementing the Run() method) I have
`using Microsoft.Tools.WindowsInstallerXml.Bootstrapper;`

Are there any hints / documentation parts that I could look through on how to change that in WiX 4 ? 
Do I need a nuget package like WixToolset.Sdk? Or for the Bootstrapping application layer rather WixToolset.BalUtil? 

If so, how should I add which packages to the Solution and which namespaces are to be used?

With WiX 3 there was always the possibility to look through your implementation of the WiX Toolset Burn Application itself, which was immensely helpful because burn documenation is a bit lacking - but now I don't really have an idea where to start exactly. Is there any coding dojo from Rob or any other link to be studied with regards to WiX 4?

**Comment by robmen** [comment]
It'll be a while before the [Deployment Dojo](https://youtube.com/playlist?list=PLDlzbQXIs18slmqmdlS10_de_Cps-QRg6) gets to building a custom BootstrapperApplication. I haven't even got to Shortcut yet! 😅  

However, I feel ya. Building up documentation around Bundles and BootstrapperApplications is something I'd like us to do at FireGiant. There's a lot of work to be done on the documentation front for v4, in general.

I've also been updating some v3 BootstrapperApplication code myself. The knowledge from v3 will serve you well but there are a lot of little differences. So be prepared. It is more than just updating the assembly references and namespaces.

I got started by looking at the [TestBA project](https://github.com/wixtoolset/wix4/blob/develop/src/test/burn/TestBA/TestBA.csproj).

**Comment by rseanhall** [comment, marked as answer]
The WiX Toolset BA isn't currently shipping, but it lives on at https://github.com/wixtoolset/wix4/blob/b500f7ae7113039c878555ec622d551854f891e2/src/test/burn/WixToolset.WixBA.
